### PR TITLE
Document configuration settings

### PR DESCRIPTION
**Pull request by lachmatt**
## Why
All of the settings supported should be documented.
## What
Update of `internal-config.md`, `advanced-config.md` and `CHANGELOG.md`.
Removal of unused configuration keys from `ConfigurationKeys`

## Tests
n/a

### CODE REVIEW

File: tracer/src/Datadog.Trace/Configuration/ConfigurationKeys.cs [hunk 1]
```diff
@@ -497,22 +497,6 @@ public static class Integrations
             public const string AnalyticsSampleRate = "SIGNALFX_TRACE_{0}_ANALYTICS_SAMPLE_RATE";
         }
 
-        /// <summary>
```
Comment (lachmatt) [inline tracer/src/Datadog.Trace/Configuration/ConfigurationKeys.cs:500]: this seems to be leftover after recent upstream pull

**Review by Kielek — APPROVED** [review]

File: docs/advanced-config.md [hunk 2]
```diff
@@ -82,3 +83,5 @@ Use these environment variables to configure the tracing library:
 | `SIGNALFX_TRACE_ROUTE_TEMPLATE_RESOURCE_NAMES_ENABLED` | ASP.NET span and resource names are based on routing configuration if applicable. | `true` |
 | `SIGNALFX_INSTRUMENTATION_ELASTICSEARCH_TAG_QUERIES` | Enable the tagging of a Elasticsearch command PostData as `db.statement`. It may introduce overhead for direct streaming users. | `true` |
 | `SIGNALFX_TRACE_DELAY_WCF_INSTRUMENTATION_ENABLED` | Enable the updated WCF instrumentation that delays execution until later in the WCF pipeline when the WCF server exception handling is established. | `false` |
+| `SIGNALFX_TRACE_KAFKA_CREATE_CONSUMER_SCOPE_ENABLED` | Enable to close consumer scope on method enter, and start a new one on method exit. | `true` |
+| `SIGNALFX_TRACE_{0}_ENABLED` | Configuration pattern for enabling/disabling specific integration, e.g in order to disable kafka integration, set `SIGNALFX_TRACE_KAFKA_ENABLED=0`|  |
```
Comment (pellared) [inline docs/advanced-config.md:87]: Should it not be `SIGNALFX_TRACE_Kafka_ENABLED=false` ? Found here: `ImmutableIntegrationSettingsCollectionTests.cs`

Also a lot of tests have the following code:

```
SetEnvironmentVariable($"SIGNALFX_TRACE_{nameof(IntegrationId.SqlClient)}_ENABLED", "false");
```

Can you double-check it?

**Review by pellared — APPROVED** [review]

File: docs/advanced-config.md (hunk 2 above)
Comment (lachmatt) [reply docs/advanced-config.md:87]: double-checked

File: docs/CHANGELOG.md [hunk 3]
```diff
@@ -30,6 +30,14 @@ and this repository adheres to [Semantic Versioning](https://semver.org/spec/v2.
 ### Breaking changes
 
 - There is no support for .NET older than .NET 4.6.2.
+- Rename `SIGNALFX_DOTNET_TRACER_CONFIG_FILE` configuration to `SIGNALFX_TRACE_CONFIG_FILE`.
+- Rename `SIGNALFX_PROPAGATOR` configuration to `SIGNALFX_PROPAGATORS`.
+- Rename `SIGNALFX_TRACE_GLOBAL_TAGS` configuration to `SIGNALFX_TAGS`.
+- Rename `SIGNALFX_TRACING_ENABLED` configuration to `SIGNALFX_TRACE_ENABLED`
+- Rename `SIGNALFX_ASPNET_TEMPLATE_NAMES_ENABLED` configuration to `SIGNALFX_TRACE_ROUTE_TEMPLATE_RESOURCE_NAMES_ENABLED`.
```
[the remark this answers is not in the document]
Comment (lachmatt) [reply docs/CHANGELOG.md:37]: done

File: docs/advanced-config.md (hunk 2 above)
Comment (pellared) [reply docs/advanced-config.md:87]: it worked on Windows but it problaly not work for Linux. See: https://docs.microsoft.com/en-us/dotnet/api/system.environment.getenvironmentvariable?view=net-6.0

```suggestion
| `SIGNALFX_TRACE_{0}_ENABLED` | Configuration pattern for enabling/disabling a specific library instrumentation, e.g in order to disable Kafka instrumentation, set `SIGNALFX_TRACE_Kafka_ENABLED=false` | `true` |
```

Comment (pellared) [reply docs/advanced-config.md]: Addressed https://github.com/signalfx/signalfx-dotnet-tracing/pull/285/commits/a76985d0be3b7eb2d0b8bb5d939a8270843c36ee

**Pull request merged**